src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1547,98 +1547,74 @@ namespace winrt::TerminalApp::implementation
         // TODO! Remove the WSL allowing. it's trivial to insert some malicious
         // stuff into WSL, via .bash_profile, so we're not giving them the (y)
 
-        // TODO! CommandlineToArgv to get the executable from the commandline.
-        // If there's one argc, and it's parent path is %ProgramFiles%, and it
-        // ends in pwsh.exe, then it's fine.
-
-        // Also, if the path is literally
-        //   %SystemRoot%\System32\wsl.exe -d <distro name>
-        // then allow it.
-
-        // Largely stolen from _tryMangleStartingDirectoryForWSL in ConptyConnection.
-        // Find the first space, quote or the end of the string -- we'll look
-        // for wsl before that.
-        const auto terminator{ commandLine.find_first_of(LR"(" )", 1) }; // look past the first character in case it starts with "
-        const auto start{ til::at(commandLine, 0) == L'"' ? 1 : 0 };
-        const std::filesystem::path executablePath{ commandLine.substr(start, terminator - start) };
-        const auto executableFilename{ executablePath.filename().wstring() };
-
-        if (executableFilename == L"wsl" || executableFilename == L"wsl.exe")
+        // is does executablePath start with %ProgramFiles%\\PowerShell, and end
+        // with `pwsh.exe`?
+        const std::vector<std::filesystem::path> powershellCoreRoots
         {
-            // We've got a WSL -- let's just make sure it's the right one.
-            if (executablePath.has_parent_path())
-            {
-                if (executablePath.parent_path().wstring() != systemDirectory)
-                {
-                    return false; // it wasn't in system32!
-                }
-            }
-            else
-            {
-                // Unqualified WSL, this is dangerous, so return false.
-                return false;
-            }
+            // Always look in "%LOCALAPPDATA%\Microsoft\WindowsApps", which is
+            // where the pwsh.exe execution alias lives.
+            { wil::ExpandEnvironmentStringsW<std::wstring>(L"%LOCALAPPDATA%\\Microsoft\\WindowsApps") },
```

Comment:
Sorry to be a stick in the mud, but unfortunately _this path is user-controlled_.

```
(dhowett-sl) ~\src\wt-1.13 % echo hi > $Env:LOCALAPPDATA\Microsoft\WindowsApps\FUN.EXE
(dhowett-sl) ~\src\wt-1.13 % FUN
ResourceUnavailable: Program 'FUN.EXE' failed to run: The specified executable is not a valid application for this OS platform.
```